plugins/MultiSites/Controller.php

```diff
@@ -40,6 +45,34 @@ public function standalone()
         return $this->getSitesInfo($isWidgetized = true);
     }
 
+    public function getAllWithGroups()
+    {
+        Piwik::checkUserHasSomeViewAccess();
```

Comment:
All this is in a controller and not API. It is not an API method as it only does some formatting etc for our dashboard widget.